pkg/collector/skipper_collector.go

```diff
@@ -128,26 +134,65 @@ func getWeights(ingressAnnotations map[string]string, backendAnnotations []strin
 	return 0.0, errBackendNameMissing
 }
 
-// getCollector returns a collector for getting the metrics.
-func (c *SkipperCollector) getCollector() (Collector, error) {
-	ingress, err := c.client.NetworkingV1beta1().Ingresses(c.objectReference.Namespace).Get(context.TODO(), c.objectReference.Name, metav1.GetOptions{})
-	if err != nil {
-		return nil, err
+func getRouteGroupWeight(backends []rgv1.RouteGroupBackendReference, backendName string) (float64, error) {
+	if len(backends) <= 1 {
+		return 1.0, nil
 	}
 
-	backendWeight, err := getWeights(ingress.Annotations, c.backendAnnotations, c.backend)
-	if err != nil {
-		return nil, err
+	for _, backend := range backends {
+		if backend.BackendName == backendName {
+			return float64(backend.Weight) / 100.0, nil
+		}
 	}
-	config := c.config
 
+	if backendName != "" {
```

Comment:
Is it possible for `backend.BackendName` to be empty? I don't think so.
I'd suggest moving this check to the top and invert (i.e. `if backendName == ""`) and returning earlier, don't see why to range over all backends to figure out an empty backend name was provided.

Reply:
Right, changed!